src/server/lib/getConfiguration.ts

```diff
@@ -8,7 +8,53 @@ import {
   RedisConfiguration,
   Stage,
 } from '@/server/models/Configuration';
+import { existsSync, readFileSync } from 'fs';
 import { featureSwitches } from '@/shared/lib/featureSwitches';
+import validateRateLimiterConfiguration from './rate-limit/validateConfiguration';
+import path from 'path';
+import { format } from 'util';
+
+const tryReadRateLimitConfigFile = () => {
+  const readAndParseFile = (path: string) => {
+    if (existsSync(path)) {
+      const configJson = readFileSync(path, 'utf-8');
+      if (configJson) {
+        return JSON.parse(configJson);
+      }
+    }
+    return undefined;
+  };
```

Comment:
I'm going to shard it out into its own file and export the result of the config load once so that the file read only happens when the server is started.